### PR TITLE
feat(continueWith): Add `continueWith`

### PR DESCRIPTION
**Pull request by josepot**
I've found myself needing this operator more and more lately, so I thought that maybe others could benefit from it too?

I'ts roughly the equivalent of doing:

```ts
connect(source$ =>
  merge(
    source$,
    source$.pipe(
      takeLast(1),
      mergeMap(lastVal => getNextObservable(lastVal))
    )
  )
);
```

but I find that ^^ too verbose and difficult to read, I would much rather do this, instead:

```ts
continueWith(getNextObservable)
```

Also, I'm very open to suggestions for a better name.

FTR: when I was getting started with RxJs, I actually thought that `concatMap` behaved like that :see_no_evil:.

### CODE REVIEW

**Comment by cartant** [comment]
Sounds okay to me. I'll review it later. It took me a while to realise that this wasn't the RxJS repo. 😅

File: source/operators/continueWith.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * can be found in the LICENSE file at https://github.com/cartant/rxjs-etc
+ */
+
+import { from, Observable, ObservableInput, Subscription } from "rxjs";
+
+const NO_VAL: any = {};
+
+export const continueWith = <T, O>(
+  mapper: (value: T) => ObservableInput<O>
+) => (source$: Observable<T>) =>
```
Comment (cartant) [inline source/operators/continueWith.ts]: nitpick: Anytime an API involves a function that returns an `ObservableInput`, it's worth using the `ObservedValueOf` mechanism that's in the core. See `concatMap`:

https://github.com/ReactiveX/rxjs/blob/df51b04d7ec68a72b3a4b0d69c3bb29264c72611/src/internal/operators/concatMap.ts#L6-L8

If this is done, functions that return multiple `ObservableInput` types effect an `Observable` with a union element type, like this:

https://github.com/ReactiveX/rxjs/blob/df51b04d7ec68a72b3a4b0d69c3bb29264c72611/spec-dtslint/operators/concatMap-spec.ts#L32-L34

**Review by cartant — COMMENTED** [review]
This LGTM, but I think this could use the same mechanism and type that's used in the core to handle functions that return multiple `ObservableInput` types via multiple branches.

File: source/operators/continueWith.ts (hunk 1 above)
Comment (josepot) [reply source/operators/continueWith.ts]: Nice! Thanks for letting me know that! I just updated the PR, please let me know if there is anything else that could be improved. Thanks!

**Review by cartant — APPROVED** [review]

**Pull request merged**